### ISSUE TITLE
purs bundle eliminates functions called locally

### ISSUE THREAD

**Issue by jamesdbrock** [purescript/purescript#4181]
## Description

PR #4044 seems to have introduced a regression in the bundler.

As @ozkutuk reported in https://github.com/purescript-contrib/purescript-arraybuffer/issues/34 , in version 0.14.4 `purs bundle` started eliminating functions from a `.js` file which are called locally in the `.js` file at “load time.” (What do we call that time at which the `.js` file is evaluated? Is it “load time”? I don't know).

## Expected behavior

This commit on my fork of __purescript__ introduces a test which I believe should pass if the bundler is working properly.

https://github.com/jamesdbrock/purescript/commit/52dd3a5881468d688263881cefc6928e462cbe0b

Instead the test fails with `purescript>        ReferenceError: localFunction is not defined`.

## To Reproduce

```
git clone https://github.com/jamesdbrock/purescript.git
cd purescript
git checkout bundle-function
stack test --test-arguments="--match FunctionDeclaration.purs"
```

## PureScript version

0.14.4

**Comment by ozkutuk** [comment]
Inspecting the bundler code, I think I understood why this happens. I am working on fixing this.

**Comment by JordanMartinez** [comment]
I believe this is resolved now that #4255 is merged.

**Comment by rhendric** [comment]
Heh, yeah, let's close all of these.
